src/utils/spid.ts

```diff
@@ -0,0 +1,160 @@
+import { QueueClient } from "@azure/storage-queue";
+import { format as dateFnsFormat } from "date-fns";
+import { isLeft, tryCatch2v } from "fp-ts/lib/Either";
+import {
+  fromEither,
+  fromNullable,
+  isNone,
+  Option,
+  tryCatch
+} from "fp-ts/lib/Option";
+import * as t from "io-ts";
+import { UTCISODateFromString } from "italia-ts-commons/lib/dates";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import {
+  FiscalCode,
+  IPString,
+  NonEmptyString,
+  PatternString
+} from "italia-ts-commons/lib/strings";
+import { DOMParser } from "xmldom";
+import { log } from "./logger";
+
+const SAML_NAMESPACE = {
+  ASSERTION: "urn:oasis:names:tc:SAML:2.0:assertion",
+  PROTOCOL: "urn:oasis:names:tc:SAML:2.0:protocol"
+};
+
+export const getFiscalNumberFromPayload = (
+  doc: Document
+): Option<FiscalCode> => {
+  return fromNullable(
+    doc.getElementsByTagNameNS(SAML_NAMESPACE.ASSERTION, "Attribute")
+  )
+    .mapNullable(collection => {
+      return tryCatch(() => {
+        return Array.from(collection).filter(
```

Comment:
if only one is returned use `find` instead of `filter`

Reply:
fixed in #d1f1f6c